.github/workflows/check_python_dependencies.yml

```diff
@@ -0,0 +1,57 @@
+name: Check Python Dependencies
+
+on:
+  pull_request:
+
+defaults:
+  run:
+    shell: bash # strict bash
+
+jobs:
+  check_dependencies:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.8'
```

Comment:
I intentionally did that because i believe we have not added support of 3.12 in edx-platform yet, 
if we try to update the python-version to 3.12 it results error 

`Traceback (most recent call last):
  File "/home/runner/work/edx-platform/edx-platform/scripts/find_python_dependencies.py", line 8, in <module>
    import requirements
  File "/opt/hostedtoolcache/Python/3.[12](https://github.com/openedx/edx-platform/actions/runs/8811123770/job/24184575317?pr=34549#step:10:13).3/x64/lib/python3.12/site-packages/requirements/__init__.py", line 19, in <module>
    from .parser import parse
  File "/opt/hostedtoolcache/Python/3.12.3/x64/lib/python3.12/site-packages/requirements/parser.py", line 23, in <module>
    from .requirement import Requirement
  File "/opt/hostedtoolcache/Python/3.12.3/x64/lib/python3.12/site-packages/requirements/requirement.py", line 24, in <module>
    from pkg_resources import Requirement as Req
ModuleNotFoundError: No module named 'pkg_resources'`

The reason for the above error is 

> [gh-95299](https://github.com/python/cpython/issues/95299): Do not pre-install setuptools in virtual environments created with [venv](https://docs.python.org/3/library/venv.html#module-venv). This means that distutils, setuptools, pkg_resources, and easy_install will no longer available by default; to access these run pip install setuptools in the [activated](https://docs.python.org/3/library/venv.html#venv-explanation) virtual environment.

> https://docs.python.org/3/whatsnew/3.12.html

Reply:
I have updated it.